bin/propolis-server/src/lib/lib.rs

```diff
@@ -1,10 +1,21 @@
+// always present
 pub mod config;
+pub mod mock_server;
+mod serial;
+#[cfg_attr(feature = "mock-only", allow(unused))]
+mod spec;
+
+#[cfg(not(feature = "mock-only"))]
 mod initializer;
+#[cfg(not(feature = "mock-only"))]
 mod migrate;
-mod serial;
+#[cfg(not(feature = "mock-only"))]
 pub mod server;
-mod spec;
+#[cfg(not(feature = "mock-only"))]
 mod stats;
+#[cfg(not(feature = "mock-only"))]
 mod vcpu_tasks;
+#[cfg(not(feature = "mock-only"))]
 mod vm;
+#[cfg(not(feature = "mock-only"))]
 pub mod vnc;
```

Comment:
We already have `cfg-if` in tree so might as well use it here to get rid of all the duplicate cfg's here.